test/test.dart

```diff
@@ -265,3 +278,14 @@ Future<void> baasTest(
 extension RealmObjectTest on RealmObject {
   String toJson() => realmCore.objectToString(this);
 }
+
+extension DateFormat on DateTime {
+  String toRealmString() {
+    final utc = toUtc();
+    // This is kind of silly, but Core serializes negative dates as -003-01-01 12:34:56
+    final utcYear = utc.year < 0 ? '-${utc.year.abs().toString().padLeft(3, '0')}' : utc.year.toString().padLeft(4, '0');
```

Comment:
Weird, and why pad 3? Just curious

Reply:
I have no idea - I believe it was likely overlooked by the core team.